documentation/src/main/asciidoc/cheat-sheets/cache-configuration.adoc

```diff
@@ -0,0 +1,414 @@
+  </indexing>
+</distributed-cache>
+----
+
+Index writer:: The index writer constructs an index composed of one or more segments (sub-indexes) that can be merged over time to improve performance.
++
+[source,xml]
+----
+<distributed-cache>
+  <indexing storage="filesystem" path="${java.io.tmpdir}/baseDir">
+    <index-writer commit-interval="2000"
+                  low-level-trace="false"
+                  max-buffered-entries="32"
+                  queue-count="1"
+                  queue-size="10000"
+                  ram-buffer-size="400"
+                  thread-pool-size="2">
+      <index-merge calibrate-by-deletes="true"
+                   factor="3"
+                   max-entries="2000"
+                   min-size="10"
+                   max-size="20"/>
+    </index-writer>
+    <!-- Additional indexing configuration goes here. -->
+  </indexing>
+</distributed-cache>
+----
+
+== Authorization
+Secure your deployment by restricting user access to data.
+
+.Default set of roles
+* Observer
+* Application
+* Admin
+* Monitor
+* Deployer
++
+[source,xml]
+----
+<distributed-cache>
+	<security>
+        <!-- Multiple roles can be used -->
+		<authorization enabled="true"
+                  roles="observer application admin monitor"/>
+	</security>
+</distributed-cache>
+----
+
+== Backups
+Define backup locations for cache data and modify state transfer properties.
+
+Backup strategy:: To back up data to a different cluster, {brandname} can use either a synchronous or asynchronous strategy.
+{brandname} performs conflict resolution with the asynchronous backup strategy.
+
+Remote site:: Specifies the name of the remote site that backs up data to the local cache.
+Remote cache:: Specifies the name of the remote cache that uses the local cache as a backup.
+Timeout:: Specifies timeout, in milliseconds, for synchronous and asynchronous backup operations.
+
+[cols="a,a", options="header"]
+|===
+| Cluster LON
+| Cluster NYC
+
+|
+[source,xml]
+----
+<distributed-cache>
+  <backups>
+    <backup site="NYC"
+            strategy="ASYNC"
+            timeout="10000" />
+  </backups>
+</distributed-cache>
+----
+
+|
+[source,xml]
+----
+<distributed-cache name="eu-customers">
+  <backups>
+    <backup site="LON"
+            strategy="ASYNC" />
+  </backups>
+    <!-- Defines the local cache as a backup for a remote cache with a different name. -->
+  <backup-for remote-cache="customers"
+              remote-site="LON" />
+</distributed-cache>
+----
+
+|===
+
+== Transactional
+
+.Transaction mode
+Configure the mode that {brandname} uses when carrying out transactions to ensure the cache state is consistent.
+
+NON_XA:: Cache will enlist within transactions as a `javax.transaction.Synchronization`.
+NON_DURABLE_XA:: Cache will enlist within transactions as a `javax.transaction.xa.XAResource`, without recovery.
+FULL_XA:: Cache will enlist within transactions as a `javax.transaction.xa.XAResource`, with recovery.
+
```

Comment:
Tip: you'll need a transactional cache, at least NON-XA, to ensure consistency for conditional operations like putIfAbsent because internally it needs to combine a get and conditional put operation. The logfile will show warn messages if such operations are used for non transactional caches.